src/main/java/org/opensearch/ad/task/ADTaskCacheManager.java

```diff
@@ -606,21 +783,47 @@ private ADHCBatchTaskCache getExistingHCTaskCache(String detectorId) {
      * @param entities list of entities
      */
     public void addPendingEntities(String detectorId, List<String> entities) {
-        getHCTaskCache(detectorId).addEntities(entities);
+        getOrCreateHCTaskCache(detectorId).addEntities(entities);
     }
 
-    private ADHCBatchTaskCache getHCTaskCache(String detectorId) {
+    private ADHCBatchTaskCache getOrCreateHCTaskCache(String detectorId) {
         return hcTaskCaches.computeIfAbsent(detectorId, id -> new ADHCBatchTaskCache());
     }
 
+    /**
+     * Check if there is any HC task running on current node.
+     * @param detectorId detector id
+     * @return true if find detector id in any entity task or HC cache
```

Comment:
Would you need to check the task states as well? Or only running tasks will be in the cache?

Reply:
Only running tasks will be in cache.